core/src/main/java/com/linecorp/armeria/server/file/HttpFile.java

```diff
@@ -177,6 +177,33 @@ static HttpFileBuilder builder(HttpData data, long lastModifiedMillis) {
                 .autoDetectedContentType(false); // Can't auto-detect because there's no path or URI.
     }
 
+    /**
+     * Returns a new {@link HttpFileBuilder} that builds an {@link HttpFile} from the specified
+     * {@link URL}. Supports file: and jar:file: protocol
+     */
+    static HttpFileBuilder builder(URL url) {
+        requireNonNull(url, "url");
+        if (url.getPath().endsWith("/")) {
+            // Non-existent resource.
+            return new NonExistentHttpFileBuilder();
+        }
+
+        // Convert to a real file if possible.
+        if ("file".equals(url.getProtocol())) {
+            File f;
+            try {
+                f = new File(url.toURI());
+            } catch (URISyntaxException ignored) {
+                f = new File(url.getPath());
+            }
+
+            return builder(f.toPath());
+        } else if ("jar".equals(url.getProtocol())) {
+            return new ClassPathHttpFileBuilder(url);
+        }
+        throw new UnsupportedOperationException("Not supported URL protocol " + url.getProtocol());
```

Comment:
Is it better to raise an `IllegalArgumentException`?

Reply:
Sure